languages/javascript/exercises/concept/array-transformations/example.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Double every card in the deck
+ *
+ * @param {number[]} deck
+ *
+ * @returns {number[]} deck with every card doubled
+ */
+export function seeingDouble(deck) {
+  return deck.map((card) => card * 2)
+}
+
+/**
+ *  Creates triplicates of every 3 found in the deck
+ *
+ * @param {number[]} deck
+ *
+ * @returns {number[]} deck with triplicate 3s
+ */
+export function threeOfEachThree(deck) {
+  return deck.reduce((newDeck, card) => {
+    if (card === 3) {
+      newDeck.push(3, 3, 3)
+    } else {
+      newDeck.push(card)
+    }
+    return newDeck
+  }, [])
+}
+
+/**
+ * Removes every card from the deck but the middle two
+ * Assumes a deck is always 10 cards
+ *
+ * @param {number[]} deck of 10 cards
+ *
+ * @returns {number[]} deck with only two middle cards
+ */
+export function middleTwo(deck) {
+  return deck.slice(4, 5)
+}
+
+/**
+ * Moves the outside two cards to the middle
+ *
+ * @param {number[]} deck with 10 cards
+ *
+ * @returns {number[]} transformed deck
+ */
+
+export function sandwichTrick(deck) {
+  const firstCard = deck.shift()
+  const lastCard = deck.pop()
+  return deck.splice(3, 0, lastCard, firstCard)
+}
+
+/**
+ * Removes every card from the deck except 2s
+ *
+ * @param {number[]} deck
+ *
+ * @returns {number[]} deck with only 2s
+ */
+export function twoIsSpecial(deck) {
+  return deck.filter((card) => card === 2)
```

Comment:
@yyyc514 Regarding your question how well-designed the example code should be, the general rule is to have the example be the most idiomatic solution you can write based on what the student has learnt in the exercise's prerequisites. In this particular case, I think in this case I would just have the magic numbers, but you could argue it either way. @SleeplessByte any preference?

Reply:
This might not of been the best example.  What about line 39 where at first glance the 4 and 5 have absolutely zero meaning to the reader. ￼

Reply:
In reality, that one can be made more general purpose by removing the magic numbers.

```javascript
const middle = Math.floor(deck.length / 2)
return deck.slice(middle,  middle + 1)
```

Reply:
This remains something of a delicate problem, where there is no definitive right or wrong in most cases. These are the rules according to [the docs](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#example-implementation-file):

- The implementation must be idiomatic.
- The code should be as simple as possible.
- Only use language features introduced by the exercise or its prerequisites (and their prerequisites, and so on).
- The example file is not shown to the student when doing in-browser coding and is not downloaded to the student's file system when using the CLI.

So as long as the example fits those rules, it is valid.

Reply:
> The code should be as simple as possible.

Idiomatic can definitely contradict with this, as is the case with this specific question.

Reply:
> Idiomatic can definitely contradict with this, as is the case with this specific question.

Correct. We should probably make the instructions a bit more clear.

Reply:
So do we want the simple version of the more flexible one? :)  Someone let me know.